test/SILOptimizer/remove_unused_global_vars.swift

```diff
@@ -0,0 +1,28 @@
+// RUN: %target-swift-frontend -O -emit-sil %s | %FileCheck %s
+
+// CHECK-NOT: sil_global private [let] {{.*}}unused1{{.*}}
+private let unused1 = 0
+// CHECK-NOT: sil_global private {{.*}}unused2{{.*}}
+private var unused2 = 42
+// CHECK: sil_global private [let] @${{.*}}used1{{.*}} : $Int
+private let used1 = 0
+// CHECK: sil_global private @${{.*}}used2{{.*}} : $Int
+private var used2 = 0
+
+// CHECK: sil_global [let] @${{.*}}unused3{{.*}} : $Int
+public let unused3 = 0
+// CHECK: sil_global @${{.*}}unused4{{.*}} : $Int
+public var unused4 = 0
+
+// These should only be optimized with -wmo.
```

Comment:
Does this already work? I would expect they get Hidden linkage not Private and wouldn't be caught by the check above.

Reply:
It didn't use to work. After applying your above comment it now works. I'll add tests for this too.

Reply:
Great!